From 0b464dc51d8e58b61bcef47db1ef7fa9d899f3b9 Mon Sep 17 00:00:00 2001
Message-Id: <patch-4.6-0b464dc51d8-20210615T161330Z-avarab@gmail.com>
In-Reply-To: <cover-0.6-00000000000-20210615T161330Z-avarab@gmail.com>
References: <cover-0.6-00000000000-20210615T161330Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 15 Jun 2021 17:53:14 +0200
Subject: [PATCH 4/6] fast-import doc: change "bob" in an example to "file.txt"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The example added in e7e5170f804 (Update fast-import documentation to
discuss crash reports, 2008-02-14) is a bit confusing in that we're
referring to a "bob" when we really just need a placeholder name for a
file that has bad mode bits, let's use "file.txt" instead.

Let's also use "<<-" in the here-doc so this'll work if the reader
copies this from e.g. a tab-indented manual page, none of the content
needs leading whitespace, so that won't break anything if the content
isn't indented.

The fast-import error message and other output has also changed
slightly since 2008, let's update the relevant parts of it, while
retaining the original PID, time etc. of the 2007-era example.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Documentation/git-fast-import.txt | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/Documentation/git-fast-import.txt b/Documentation/git-fast-import.txt
index 39cfa05b28b..ddaa8e5d755 100644
--- a/Documentation/git-fast-import.txt
+++ b/Documentation/git-fast-import.txt
@@ -1277,7 +1277,7 @@ must be applied manually if the update is needed.
 An example crash:
 
 ====
-	$ cat >in <<END_OF_INPUT
+	$ cat >in <<-END_OF_INPUT
 	# my very first test commit
 	commit refs/heads/master
 	committer Shawn O. Pearce <spearce> 19283 -0400
@@ -1289,12 +1289,13 @@ An example crash:
 	data <<EOF
 	.gitignore
 	EOF
-	M 777 inline bob
+	M 777 inline file.txt
 	END_OF_INPUT
 
 	$ git fast-import <in
-	fatal: Corrupt mode: M 777 inline bob
+	fatal: Corrupt mode: M 777 inline file.txt
 	fast-import: dumping crash report to .git/fast_import_crash_8434
+	Unpacking objects: 100% (1/1), 32 bytes | 32.00 KiB/s, done.
 
 	$ cat .git/fast_import_crash_8434
 	fast-import crash report:
@@ -1302,7 +1303,7 @@ An example crash:
 	    parent process     : 1391
 	    at Sat Sep 1 00:58:12 2007
 
-	fatal: Corrupt mode: M 777 inline bob
+	fatal: Corrupt mode: M 777 inline file.txt
 
 	Most Recent Commands Before Crash
 	---------------------------------
@@ -1313,7 +1314,7 @@ An example crash:
 	  data <<EOF
 	  M 644 inline .gitignore
 	  data <<EOF
-	* M 777 inline bob
+	* M 777 inline file.txt
 
 	Active Branch LRU
 	-----------------
@@ -1334,6 +1335,9 @@ An example crash:
 	  last pack   :
 
 
+	Marks
+	-----
+
 	-------------------
 	END OF CRASH REPORT
 ====
-- 
2.32.0.555.g0268d380f7b

